src/stirling/source_connectors/socket_tracer/protocols/dns/stitcher.cc

```diff
@@ -237,7 +245,33 @@ RecordsWithErrorCount<Record> StitchFrames(std::deque<Frame>* req_frames,
 
   resp_frames->clear();
 
-  return {entries, error_count};
+  if (include_respless_dns_requests) {
+    // After the external loop's lifecycle comes to an end we end up with the request deque
+    // having only those request frames which have not been consumed yet i.e. consumed = false
+    // so essentially these are the requests which could not be matched with any response frame.
+    // Hence we iterate over this request deque, add a default response to it, make a record and
+    // append those records at the end of the entries vector.
+    auto it = req_frames->begin();
+    while (it != req_frames->end()) {
+      if (!(*it).consumed) {
+        auto current_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+                                std::chrono::system_clock::now().time_since_epoch())
+                                .count();
```

Comment:
Getting the clock can become expensive when inside a loop. Can you get the current_time a single time outside the loop and use the fixed value inside the loop. While not strictly as accurate, it should be more than sufficient for this use case.